barretenberg/cpp/src/barretenberg/stdlib/honk_verifier/ultra_recursive_verifier.cpp

```diff
@@ -127,7 +127,39 @@ UltraRecursiveVerifier_<Flavor>::AggregationObject UltraRecursiveVerifier_<Flavo
     pairing_points[1] = pairing_points[1].normalize();
     // TODO(https://github.com/AztecProtocol/barretenberg/issues/995): generate recursion separator challenge properly.
     agg_obj.aggregate(pairing_points, recursion_separator);
-    return agg_obj;
+    Output output;
+    output.agg_obj = std::move(agg_obj);
+
+    // Extract the IPA claim from the public inputs
+    // Parse out the nested IPA claim using key->ipa_claim_public_input_indices and runs the native IPA verifier.
+    if constexpr (HasIPAAccumulatorFlavor<Flavor>) {
+        const auto recover_fq_from_public_inputs = [](std::array<FF, 4>& limbs) {
+            for (size_t k = 0; k < 4; k++) {
```

Comment:
Magic number 4--if we ever added a number this would break; can extract from builder.